calendar/src/main/java/com/toggl/calendar/domain/CalendarItemsSelector.kt

```diff
@@ -0,0 +1,26 @@
+package com.toggl.calendar.domain
+
+import com.toggl.environment.services.calendar.CalendarEvent
+import com.toggl.models.domain.TimeEntry
+import org.threeten.bp.OffsetDateTime
+
+fun calendarItemsSelector(
+    calendarLayoutCalculator: CalendarLayoutCalculator,
+    timeEntries: Map<Long, TimeEntry>,
+    calendarEvents: List<CalendarEvent>,
+    date: OffsetDateTime
+): List<CalendarItem> {
+    val localDate = date.toLocalDate()
+    val filteredTimeEntries = timeEntries.filterValues {
+        val endTime = it.startTime + it.duration
+        it.startTime.toLocalDate() == localDate && (endTime == null || endTime.toLocalDate() == localDate)
+    }
+    val filteredEvents = calendarEvents.filter {
+        val endTime = it.startTime + it.duration
+        it.startTime.toLocalDate() == localDate && (endTime == null || endTime.toLocalDate() == localDate)
+    }
```

Comment:
This better hides the unimportant details so the reader can focus on higher-level concepts and only dive deeper when necessary. Not sure about the `isOnDate` name though.
```suggestion
    val localDate = date.toLocalDate()
    fun isOnDate(startTime: OffsetDateTime, endTime: OffsetDateTime?) =
        startTime.toLocalDate() == localDate && (endTime == null || endTime.toLocalDate() == localDate)

    val filteredTimeEntries = timeEntries.filterValues { isOnDate(it.startTime, it.startTime + it.duration) }
    val filteredEvents = calendarEvents.filter { isOnDate(it.startTime, it.startTime + it.duration) }
```

Reply:
`startsOnDate` is a better name, since it better describes the logic used to include events/time entries in the calendar

Reply:
`startsOnDate` does not describe de actual behaviour here. We are filtering the time entries that **Start And End** on this date.

`isOnDate` or maybe `isOnTheSameDay` would describe it better.